### ISSUE TITLE
Generated files don't load via the local preview url

### ISSUE THREAD

**Issue by tmharryfrancis** [Shopify/shopify-cli#1530]
# Issue summary

Write a short description of the issue here ↓

For files generated by Shopify (eg. theme.css.liquid), hot Reloading doesn't seem to work. I'm assuming this is because you would need to somehow parse the liquid locally, as files are served from the local assets folder?

If I look at the site on the external preview URL, it works as expected.

## Expected behavior

What do you think should happen?

When editing the theme.css.liquid file, the changes should be reflected on my local preview URL. This means all generated assets should be served from the external store (where they are generated), rather than locally (where they don't exists/get updated).

## Actual behavior

What actually happens?

When I edit the theme.css.liquid file, non of the updates get reflected. The only way to see them is by re-running the serve command (which I'd have to do on every edit, so doesn't seem feasible), or by viewing the site on the external preview URL (which is a temporary fix, as if you go back to the local preview URL the changes haven't changed).

## Steps to reproduce the problem

1. Setup the CLI with a theme that has a theme.css.liquid file
2. Reference the theme.css.liquid file in the theme
3. Run the shopify serve command
4. Look at the site via the 'http://127.0.0.1:9292' (or similar) url
4. Edit the theme.css.liquid file
5. Notice no changes get made

## Specifications

- App type: N/A
- Operating System: macOS Big Sur
- Shell: zsh
- Ruby version (ruby -v): ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [arm64-darwin20]

**Comment by pepibumur** [comment]
👋🏼 @tmharryfrancis 
It's Pedro from Shopify. Apologies for the inconveniences this is causing to your experience using the platform.
To debug and fix the issue I'll need some extra guidance from you to reproduce the issue. These are the steps that I followed:

1. Create a theme with `shopify theme init`
2. Run `shopify theme serve`
3. Open `http://127.0.0.1:9292`
4. Modify a liquid file like `rich-text.liquid`
5. The browser reloads automatically.

I wonder if it might have something to do with the file being modified. Also extra context it'd be great to have from you:
- What version of the CLI are you using. You can get it by running `shopify version`.
- When previewing the theme on the browser, do you see any error in the Javascript console? If so, do you mind sharing it?

**Comment by macournoyer** [comment]
Hey @tmharryfrancis!

Oh you're right! It doesn't work because we refresh before it gets updated on the external store (where they are generated), like you noted.

A manual refresh should work. Can you confirm? If so, I think adding a small delay might fix this edge case.

**Comment by tmharryfrancis** [comment]
Hey @macournoyer,

Even after a refresh, it doesn't work for me. To get it to update I have to re-run the serve command.

Did you try the manual refresh and get it working, or was it assumed?

Thanks, Harry.

**Comment by hpawe01** [comment]
I also noticed this problem with the generated javascript file `scripts.js`. I have a `{% javascript %}` part in one of my components and if I make changes there it takes some time until the changes are visible in the local and preview url and customizer. Even re-running the serve command doesn't fix this. I guess this file is cached somehow?

The only way I found so far to make this work, is to change something else in the components DOM. This seems to also trigger a recreation of the `scripts.js` file.
